src/components/DataSource/independent.ts

```diff
@@ -0,0 +1,42 @@
+import { ValueRef } from '@holoflows/kit/es'
+import Services from '../../extension/service'
+import { PersonaArrayComparer } from '../../utils/comparer'
+import { MessageCenter } from '../../utils/messages'
+import { Persona } from '../../database'
+import { useValueRef } from '../../utils/hooks/useValueRef'
+import { PluginMessageCenter } from '../../plugins/PluginMessages'
+import { currentEthereumNetworkSettings } from '../../plugins/Wallet/network'
+import { WalletRecord, ERC20TokenRecord } from '../../plugins/Wallet/database/types'
+
+const independentRef = {
+    myPersonasRef: new ValueRef<Persona[]>([], PersonaArrayComparer),
+    // TODO: typeonly import
+    walletTokenRef: new ValueRef<[(WalletRecord & { privateKey: string })[], ERC20TokenRecord[]]>([[], []]),
+}
+
+{
+    const ref = independentRef.myPersonasRef
+    query()
+    MessageCenter.on('personaUpdated', query)
+    function query() {
+        Services.Identity.queryMyPersonas().then(p => (ref.value = p))
+    }
+}
+
+{
+    const ref = independentRef.walletTokenRef
+    query()
+    PluginMessageCenter.on('maskbook.wallets.update', query)
+    currentEthereumNetworkSettings.addListener(query)
+    function query() {
+        Services.Plugin.invokePlugin('maskbook.wallet', 'getWallets').then(x => (ref.value = x))
+    }
+}
+
+export function useMyPersonas() {
+    return useValueRef(independentRef.myPersonasRef)
```

Comment:
why move persona here🤔

Reply:
It was independent so moved here...

Reply:
Maybe we should categorize them by maskbook or plugin

Reply:
are we still maintaining a plugin system?

Reply:
yes, we should distinguish them at the term level? 🤔

Reply:
If the wallet is not a plugin (in future), I'm ok with this change

Reply:
> If the wallet is not a plugin (in future), I'm ok with this change

ok